src/test/common/test_random.cc

```diff
@@ -0,0 +1,224 @@
+  auto z = gen_fn();
+  gen_fn.seed(42);   // re-seed
+ 
+  // Mostly to swallow "unused variable" warnings:
+  ASSERT_TRUE(none_equal(a, b, c, d, e, x, y, z));
+}
+
+TEST(util, test_random)
+{
+  /* The intent of this test is not to formally test random number generation, 
+  but rather to casually check that "it works" and catch regressions: */
+ 
+  // The default overload should compile:
+  ceph::util::randomize_rng();
+ 
+  {
+    int a = ceph::util::generate_random_number();
+    int b = ceph::util::generate_random_number();
+ 
+    /* Technically, this can still collide and cause a false negative, but let's 
+    be optimistic: */
+    if (std::numeric_limits<int>::max() > 32767) {
+       ASSERT_GT(a, -1);
+       ASSERT_GT(b, -1);
+ 
+       ASSERT_NE(a, b);
+     }
+  }
+ 
+  {
+  auto a = ceph::util::generate_random_number(1, std::numeric_limits<int>::max());
+  auto b = ceph::util::generate_random_number(1, std::numeric_limits<int>::max());
+ 
+  if (std::numeric_limits<int>::max() > 32767) {
+     ASSERT_GT(a, 0);
+     ASSERT_GT(b, 0);
+ 
+     ASSERT_NE(a, b);
+   }
+  }
+ 
+  for (auto n = 100000; n; --n) {
+     constexpr int min = 0, max = 6;
+     int a = ceph::util::generate_random_number<min, max>();
+     ASSERT_GT(a, -1);
+     ASSERT_LT(a, 7);
+   }
+ 
+  // Multiple types (integral):
+  {
+    int min = 0, max = 1;
+    type_check_ok(min, max);
+  }
+ 
+  {
+    long min = 0, max = 1l;
+    type_check_ok(min, max);
+  }
+ 
+  // Multiple types (floating point):
+  {
+    double min = 0.0, max = 1.0;
+    type_check_ok(min, max);
+  }
+ 
+  {
+    float min = 0.0, max = 1.0;
+    type_check_ok(min, max);
+  }
+ 
+  // min > max should not explode:
+  {
+    float min = 1.0, max = 0.0;
+    type_check_ok(min, max);
+  }
+}
+
+TEST(util, test_make_random_function)
+{
+  constexpr auto max_int   = std::numeric_limits<int>::max();
+ 
+  {
+    auto rng_i = ceph::util::make_random_number_function<1, max_int - 1>();
+ 
+    int x = rng_i(), y = rng_i();
+ 
+    ASSERT_GT(x, 0);
+    ASSERT_LT(x, max_int);
+    ASSERT_GT(y, 0);
+    ASSERT_LT(y, max_int);
+    ASSERT_NE(x, y);
+  }
+  
+  // User may specify engine:
+  {
+    auto rng_i = ceph::util::make_random_number_function<1, 10, std::mt19937_64>();
+    int x = rng_i();
+    ASSERT_GT(x, 0); 
+    ASSERT_LT(x, 10); 
+  }
+ 
+  // User specified seed:
+  {
+    auto rng_i = ceph::util::make_random_number_function<1, 10, 42>();
+    int x = rng_i();
+    ASSERT_GT(x, 0); 
+    ASSERT_LT(x, 10); 
+  }
+}
+
+TEST(util, test_random_class_interface)
+{
+  ceph::util::random_number_generator<int> rng;
+ 
+  // Other ctors:
+  {
+    ceph::util::random_number_generator<int> rng(1234);   // seed
+  }
+ 
+  {
+    auto a = rng();
+    auto b = rng();
+ 
+    // Technically can fail, but should "almost never" happen:
+    ASSERT_NE(a, b);
+  }
+ 
+  {
+    auto a = rng(10);
+    ASSERT_LE(a, 10);
+    ASSERT_GE(a, 0);
+  }
+ 
+  {
+    auto a = rng(10.0);
```

Comment:
no, `rng` is an integer random number generator, and `a` will be an integer. this test is not verifying your expectation. instead, i think it demonstrates a misuse of this utility.